### DISCUSSION TITLE
What is the rationale for time stamped migration files?

### DISCUSSION THREAD

**Discussion by MaxKerney** [supabase/supabase#19047] [Questions]
I'm just starting out learning about database development with Supabase and I'm a bit confused by the [recommended practice](https://supabase.com/docs/guides/cli/local-development#database-migrations) of creating lots of time stamped migration files when working locally, one for each change to the database. As the database is being rebuilt from scratch after every reset, would it not be more efficient to just write your SQL code in one "database build" file (or perhaps a few, e.g., one for specifying tables, one for specifying roles, etc.) and just edit this file/these files when you want to make changes or additions, with any such changes being tracked via Git commits? As I said, I'm new to all of this, so there's probably something I'm missing. I thought I'd ask the question so that somebody could set me straight!

**Comment by EskelCz** [comment]
Yes! That's just what I came here to ask. When working with LLMs to generate database schema, I could use just single SQL file for my schema and sync from it. The source of truth being in a single text file.

ChatGPT even suggested CLI code `supabase db exec --file path/to/your/schema.sql`, but that was a hallucination. :-/
